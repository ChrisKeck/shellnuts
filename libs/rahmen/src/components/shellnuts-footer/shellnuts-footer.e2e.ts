import { test,expect } from '@playwright/test';

test.describe('shellnuts-footer', async() => {
  test.beforeEach(async({page})=>{
    await page.goto("http://localhost:3334")
  })
  test('renders', async ({page}) => {
    await expect(page).toHaveURL("http://localhost:3334/")
  });
});
