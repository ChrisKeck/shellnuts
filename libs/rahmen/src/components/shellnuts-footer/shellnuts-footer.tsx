import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'shellnuts-footer',
  styleUrl: 'shellnuts-footer.scss',
  shadow: true,
})
export class ShellnutsFooter {
  /**
   * The first name
   */
  @Prop() first: string;

  /**
   * The middle name
   */
  @Prop() middle: string;

  /**
   * The last name
   */
  @Prop() last: string;

  private getText(): string {
    return `${this.first} ${this.middle} ${this.last}`;
  }

  render() {
    return <div>Hello, World! I'm {this.getText()}</div>;
  }
}
