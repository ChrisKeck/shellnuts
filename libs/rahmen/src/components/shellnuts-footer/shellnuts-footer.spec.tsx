import { newSpecPage } from '@stencil/core/testing';
import { ShellnutsFooter } from './shellnuts-footer';

describe('shellnuts-footer', () => {
  it('renders', async () => {
    const { root } = await newSpecPage({
      components: [ShellnutsFooter],
      html: '<shellnuts-footer></shellnuts-footer>',
    });
    expect(root).toEqualHtml(`
      <shellnuts-footer>
        <mock:shadow-root>
          <div>
            Hello, World! I'm
          </div>
        </mock:shadow-root>
      </shellnuts-footer>
    `);
  });

  it('renders with values', async () => {
    const { root } = await newSpecPage({
      components: [ShellnutsFooter],
      html: `<shellnuts-footer first="Stencil" last="'Don't call me a framework' JS"></shellnuts-footer>`,
    });
    expect(root).toMatchSnapshot()
  });
});
