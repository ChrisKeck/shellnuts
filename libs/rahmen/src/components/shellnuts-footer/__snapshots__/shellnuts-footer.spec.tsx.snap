// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shellnuts-footer renders with values 1`] = `
<shellnuts-footer first="Stencil" last="'Don't call me a framework' JS">
  <mock:shadow-root>
    <div>
      Hello, World! I'm Stencil undefined 'Don't call me a framework' JS
    </div>
  </mock:shadow-root>
</shellnuts-footer>
`;
